tests/api/controllers/TokenControllerTest.php

```diff
@@ -0,0 +1,365 @@
+     */
+    protected $request;
+
+    /**
+     * @var Response $response The Phalcon HTTP response object.
+     */
+    protected $response;
+
+    /**
+     * @var TokenController $controller The TokenController mock.
+     */
+    protected $controller;
+
+    /**
+     * This method is called before the first test of this test class is run.
+     * It creates a test user.
+     */
+    public static function setUpBeforeClass(): void
+    {
+        self::createTestUser();
+    }
+
+    /**
+     * This method is called before each test. It initializes the TokenController mock.
+     *
+     * @return void
+     */
+    protected function setUp(): void
+    {
+        $this->controller = $this->getTokenControllerMock();
+    }
+
+    /**
+     * This test verifies that a valid OAuth request returns a 200 status code.
+     *
+     * @return void
+     */
+    public function testPostActionValidRequest(): void
+    {
+        $this->mockOAuthRequest();
+        $this->mockOAuthServer();
+        $response = $this->getAccessToken();
+        $this->assertEquals(200, $response->getStatusCode());
+    }
+
+    /**
+     * This test verifies that an expired token results in an unauthorized exception.
+     *
+     * @return void
+     */
+    public function testTokenExpiry(): void
+    {
+        $this->mockOAuthRequest();
+        $this->mockOAuthServer(['access_lifetime' => -1]);
+
+        $response = $this->getAccessToken();
+        $accessToken = (json_decode($response->getContent()))->access_token;
+
+        list($userControllerReflection, $userMock) = $this->getUser();
+
+        $request = new Request();
+        $request->headers = ['AUTHORIZATION' => "Bearer $accessToken"];
+        $userMock->method('getOAuthRequest')->willReturn($request);
+
+        $authorize = $userControllerReflection->getMethod('authorize');
+        $authorize->setAccessible(true);
+        $this->expectException(\Gaia\Exception\UnAuthorized::class);
+        $this->expectExceptionMessage('Invalid Token');
+        $authorize->invoke($userMock);
+    }
+
+    /**
+     * This test verifies that the remember me functionality sets the correct expiration time.
+     *
+     * @return void
+     */
+    public function testSetRememberMe(): void
+    {
+        $this->mockOAuthRequest(true);
+        $this->mockOAuthServer();
+        $this->getAccessToken();
+        $user = \Gaia\MVC\Models\User::findFirstByUsername('testUserOAuth');
+        $this->assertGreaterThan(gmdate('Y-m-d H:i:s'), $user->rememberMe);
+    }
+
+    /**
+     * This test verifies that the remember me functionality expires correctly.
+     *
+     * @return void
+     */
+    public function testRememberMeExpiry(): void
+    {
+        $this->mockOAuthRequest(false);
+        $this->mockOAuthServer();
+        $this->getAccessToken();
+        $user = \Gaia\MVC\Models\User::findFirstByUsername('testUserOAuth');
+        $this->assertLessThan(gmdate('Y-m-d H:i:s'), $user->rememberMe);
```

Comment:
Same goes here..

Reply:
Test case updated with the instructions. Please review.